docs/user-guide/lazy-loading.md

```diff
@@ -0,0 +1,88 @@
+# Lazy Loading
+
+> [!WARNING]
+> This is an experimental option and may not work as expected with all plugins.
+> The API may change without notice. > Please report any issues you encounter.
+
+When the lazy loading functionality is enabled, the plugin's configuration is
+routed to the enabled lazy provider. Options mentioned will be in relation to
+the `${namespace}.${name}.lazyLoad` context.
+
+## Supported Lazy Loading Providers
+
+It is recommended to become familiar with lazy loading strategies supported by
+your provider in their upstream documentation and the corresponding plugin's
+support.
+
+- [Lz.n](https://github.com/nvim-neorocks/lz.n?tab=readme-ov-file#plugin-spec)
+
+## Enabling Lazy Loading Per Plugin
+
+There are a couple different ways to enable lazy loading for a particular
+plugin.
+
+- `enable` set to `true`.
+- `settings` containing some configuration.
+
+If you are just wanting to store potential configuration without enabling it,
+you can explicitly disable it setting `enable = false`.
+
+When a plugin has lazy loading enabled, the Lua configuration for your plugin
+will be passed along to the corresponding lazy provider instead. This behavior
+can be overridden by using the `settings` option that will be passed to the
+provider.
+
+For `Lz.n`, that is the `settings.after` option.
+
+## Configuring Triggers
+
+Currently, you need to define the trigger conditions in which a plugin will be
+loaded. This is done through the `settings` option.
+
+```nix
+plugins = {
+  grug-far = {
+    enable = true;
+    lazyLoad = {
+      settings = {
+        cmd = "GrugFar";
+      };
+    };
+  };
+};
+```
+
+```nix
+plugins = {
+  glow = {
+    enable = true;
+    lazyLoad.settings.ft = "markdown";
+  };
+```
+
+```nix
+plugins.toggleterm = {
+  enable = true;
+  lazyLoad = {
+    settings = {
+      cmd = "ToggleTerm";
+      keys = [
+        "<leader>tg"
+        "<leader>gg"
+      ];
+    };
+  };
+};
+```
+
+The only exception to this is `colorschemes`. In `lz-n`, we will set up the
+`colorscheme` trigger to the name of the `colorscheme` so that it is lazy loaded
+when the `colorscheme` is requested.
```

Comment:
Maybe this could be re-framed as a dedicated `## Colour schemes` section, rather than a footnote paragraph in the triggers section?

We could rephrase _"The only exception is..."_ to something more positive like _"Additionally, plugins in the `colorschemes.*` namespace will..."_ :thinking: